app/src/main/java/org/glucosio/android/activity/AddGlucoseActivity.java

```diff
@@ -41,26 +40,31 @@
 import org.glucosio.android.tools.FormatDateTime;
 import org.glucosio.android.tools.GlucosioConverter;
 import org.glucosio.android.tools.LabelledSpinner;
+import org.glucosio.android.tools.NumberFormatUtils;
 
 import java.text.DecimalFormat;
+import java.text.NumberFormat;
 import java.util.Arrays;
 import java.util.Calendar;
 
 
 public class AddGlucoseActivity extends AddReadingActivity {
 
-    static final int CUSTOM_TYPE_SPINNER_VALUE = 11;
+    private static final int CUSTOM_TYPE_SPINNER_VALUE = 11;
+
     private TextView readingTextView;
     private EditText typeCustomEditText;
     private EditText notesEditText;
     private LabelledSpinner readingTypeSpinner;
     private boolean isCustomType = false;
 
+    private final NumberFormat numberFormat = NumberFormatUtils.createDefaultNumberFormat();
```

Comment:
Just review where you are using the `numberFormat` since it is not thread-safe as in the docs:
```
Number formats are generally not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.
```

Reply:
Agree, but it is Activity class, where everything is operated on UI thread